### PR TITLE
Add debug modes to blackbox tab

### LINKED ISSUE #1502: [FR] Add Blackbox debug modes to the Blackbox tab.
Add Blackbox debug modes to the Blackbox tab. Probably as a dropdown menu. Maybe incorporate or link to[ this wiki article](https://github.com/betaflight/betaflight/wiki/Debug-Modes). And maybe order the list the same way as it's in the article, but if that prevents it from being auto generated, then go for the least amount of future maintainability. :)

Maybe make Gyro_scaled the default, instead of none? (putting this here to start a discussion, please don't disregard the while request because of this part)

I'm requesting this addition because if I ask someone to make me a BB log, they have to go into 3 different tabs to set it up in a meaningful way.

### PR DESCRIPTION
**Pull request by IvoFPV**
Fixes #1502
Adds debug mode setting in the blackbox tab. Requires MSP to go with it which will be added.

### CODE REVIEW

File: src/js/tabs/onboard_logging.js [hunk 1]
```diff
@@ -250,6 +253,85 @@ TABS.onboard_logging.initialize = function (callback) {
...
+                {text: "DUAL_GYRO_SCALED"},
+                {text: "DSHOT_RPM_ERRORS"},
+                {text: "CRSF_LINK_STATISTICS_UPLINK"},
+                {text: "CRSF_LINK_STATISTICS_PWR"},
+                {text: "CRSF_LINK_STATISTICS_DOWN"},
+                {text: "BARO"},
+                {text: "GPS_RESCUE_THROTTLE_PID"},
+                {text: "DYN_IDLE"},
+            ];
+
```
Comment (hydra) [inline src/js/tabs/onboard_logging.js]: as noted in https://github.com/betaflight/betaflight/pull/8681#pullrequestreview-273065823, if you have a 'DEBUG_MODE_COUNT' available from the firmware you can add 'UNKNOWN' for each extra item to the `debugModes` list here.  This allows new debug modes to be added to the firmware and the configurator will still work.

Comment (IvoFPV) [reply src/js/tabs/onboard_logging.js]: Updated with debug count

File: src/js/tabs/onboard_logging.js [hunk 2]
```diff
@@ -250,6 +254,89 @@ TABS.onboard_logging.initialize = function (callback) {
...
+            for (var i = 0; i < PID_ADVANCED_CONFIG.debugModeCount; i++) {
+                if (i < debugModes.length) {
+                    debugModeSelect.append(new Option(debugModes[i].text, i));
+                } else {
+                    debugModeSelect.append(new Option("UNKNOWN", i));
```
Comment (McGiverGim) [inline src/js/tabs/onboard_logging.js]: This `UNKNOWN` value must go in the messages file to be translated, like we do with other unknowns elements (like in the OSD tab). I know that here, the rest of them are usually English, but I think is a good idea to maintain the same behaviour.

Comment (IvoFPV) [reply src/js/tabs/onboard_logging.js]: Added to messages for translation.

**Comment by McGiverGim** [comment]
@hydra is that ok for you now?

**Comment by hydra** [comment]
sorry for late reply on this one @McGiverGim been a busy week yes, that is perfect now.

I did note in another PR recently that the internal debug variables in multiwii/baseflight/cleanflight/betaflight all pre-date blackbox logging and are for developers. 

The variables are not specifically for logging, but by adding this to the UI on the blackbox logging page you might think they were for that when this is not the case.  

Renaming the setting 'Blackbox debug mode' further confuses this situation.

debug variables are used in the following places:
1) internally in firmware
2) readable via msp commands
3) displayable on OSD.
4) on 'sensors' page in the configurator UI.
5) in the blackbox log data.

For a long time I thought about adding this setting to the 'sensors' page in the UI but never got round to it.  Perhaps this setting should be moved from the 'Blackbox' page to the 'Sensors' tab, or perhaps it really belongs in the catch-all 'Configuration' page and called 'Debug mode' rather than 'Blackbox debug mode'.

Perhaps the real solution is to allow the user to choose exactly what they want to log in the 4 blackbox log uint16_t placeholders from all available sources and leave 'debug mode' alone.

**Pull request merged**